File: 프로그래머스/1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.java
class Solution {
    public int[] solution(long n) {
        // 숫자를 문자열로 변환
        String str = Long.toString(n);
        
        // 문자열을 뒤집고 각 자릿수를 배열에 저장
        int[] answer = new int[str.length()];
        
        for (int i = 0; i < str.length(); i++) {
            // 뒤에서부터 각 자릿수를 추출하여 배열에 저장
            answer[i] = str.charAt(str.length() - 1 - i) - '0';
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(long long n) {
    // 자릿수 구하기
    long long tmp = n;
    int count = 0;
    while (tmp > 0) {
        count++;
        tmp /= 10;
    }
    
    // 동적 할당
    int* answer = (int*)malloc(sizeof(int) * count);

    // 뒤집어서 저장
    for (int i = 0; i < count; i++) {
        answer[i] = n % 10;
        n /= 10;
    }

    return answer;  // 배열 포인터 반환
}

